File: spec/factories/scores.rb
FactoryBot.define do
  factory :score do
    nick_name { "テストプレイヤー" }
    difficulty { "standard" }
    game_mode { "random" }
    score { 1000 }
    time_ms { nil }

    trait :random_score do
      game_mode { "random" }
      score { 1000 }
      time_ms { nil }
    end

    trait :timeattack_score do
      game_mode { "timeattack" }
      score { nil }
      time_ms { 30_000 }
    end
  end
end

File: app/frontend/lib/api/scores.ts
import { postJson } from '@/lib/http';

/**
 * スコアを保存する。
 * 旧 supabaseService.saveScore のペイロード仕様を踏襲し、
 * Rails の '/api/scores' エンドポイントへ POST する。
 */
export async function saveScore(
	nickName: string,
	score: number,
	difficulty: 'beginner' | 'standard' | 'advanced' = 'standard',
	gameMode?: 'random' | 'timeattack',
	time?: number
): Promise<{ success: boolean; error?: string }> {
	const payload: Record<string, unknown> = {
		nick_name: nickName,
		difficulty: difficulty
	};

	if (gameMode === 'timeattack') {
		payload.time = time;
		payload.game_mode = 'timeattack';
	} else {
		payload.score = score;
		payload.game_mode = gameMode || 'random';
	}

	try {
		await postJson('/api/scores', payload);
		return { success: true };
	} catch (error) {
		console.error('Error saving score:', error);
		return {
			success: false,
			error: error instanceof Error ? error.message : 'スコアの登録に失敗しました'
		};
	}
}
